### PR TITLE
PST with square root trusted setup

### PR DESCRIPTION
**Pull request by maramihali**
Implemented the polynomial optimisation excluding the MIPP

### CODE REVIEW

File: src/sqrt_pst.rs [hunk 1]
```diff
@@ -0,0 +1,273 @@
...
+      if i >> j & 1 == 1 {
+        prod = prod * b_j;
+      } else {
+        prod = prod * (Scalar::one() - b_j)
+      };
+    }
+    prod
+  }
+
+  pub fn open_q(
```
Comment (nikkolasg) [inline src/sqrt_pst.rs:134]: This function could be called only "open" and the q is computed inside ? Do we really need these two external function to call first before calling this one ?

File: src/sqrt_pst.rs [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+    let pst_proof = MultilinearPC::<I>::open(ck, q, &b);
+    timer_proof.stop();
+
+    timer_open.stop();
+
+    // TODO: add MIPP proof as return value
+    (U, pst_proof)
+  }
+
+  pub fn verify_q(
```
Comment (nikkolasg) [inline src/sqrt_pst.rs:191]: call it "verify" as consistent with the polynomial APIs

File: src/sqrt_pst.rs [hunk 3]
```diff
@@ -0,0 +1,273 @@
+use ark_bls12_377::{Bls12_377 as I, G1Affine};
```
Comment (vmx) [inline src/sqrt_pst.rs:1]: Single letter uppercase names are usually used for generics. When I read the source below, I was a bit confused where the generic `I` was defined. Then I found out it's a concrete type. If `Bls12_377` is a too long name, then perhaps just `Bls`?

Comment (vmx) [reply src/sqrt_pst.rs:1]: Or `Bls12`, this is what `blstrs` is using as type.

Comment (nikkolasg) [reply src/sqrt_pst.rs:1]: We will move this codebase to generic later on. This requires a larger effort accross the whole initial spartan codebase.
In other words, we'll have to make a "beautify and productify" pass over the whole codebase at some point.

**Comment by maramihali** [comment]
Closing this and opening a PR with the full scheme.

**Pull request closed without merging**